src/main/groovy/edu/oregonstate/mist/metaxe/health/MetaXEHealthCheck.groovy

```diff
@@ -26,15 +26,13 @@ class MetaXEHealthCheck extends HealthCheck {
             Map<String, Attributes> allAttributes
             try {
                 allAttributes = XEAppDAO.getAllAttributes(xeAppsFile)
+                if (allAttributes.isEmpty()) {
+                    throw new Exception("No data found in xeAppsFile")
+                }
             } catch (Exception e) {
                 return Result.unhealthy(e)
             }
-
-            if (allAttributes.isEmpty()) {
-                Result.unhealthy("No data found in xeAppsFile")
-            } else {
-                Result.healthy()
-            }
+            Result.healthy()
```

Comment:
Looks good. One step further, you can also combine these try-catch blocks by using [multiple catch block](https://www.tutorialspoint.com/groovy/groovy_exception_handling.htm). For example:

```groovy
try {
    File xeAppsFile = new File(xeAppsFilePath)
    if (!xeAppsFile.exists()) {
      throw new Exception("xeAppsFilePath: ${xeAppsFilePath} does not exist")
    }
    Map<String, Attributes> allAttributes
    allAttributes = XEAppDAO.getAllAttributes(xeAppsFile)
    if (allAttributes.isEmpty()) {
        throw new Exception("No data found in xeAppsFile")
    }
    return Result.healthy()
} catch (NullPointerException ignore) {
    // This shouldn't happen since a null xeAppsFilePath property in the configuration
    // file should prevent the run task from succeeding
    return Result.unhealthy("xeAppsFilePath is null")
} catch (Exception e) {
    return Result.unhealthy(e)
}
```

Reply:
It's also a good practice to [extend exception class](InvalidTermException) for different cases. It will help to improve the code readability.